drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5172,9 +5172,9 @@ Vector<uint8_t> RenderingDeviceVulkan::shader_compile_binary_from_spirv(const Ve
 		uint32_t offset = 0;
 		uint8_t *binptr = ret.ptrw();
 		binptr[0] = 'G';
-		binptr[1] = 'V';
+		binptr[1] = 'S';
 		binptr[2] = 'B';
-		binptr[3] = 'D'; // Godot vulkan binary data.
+		binptr[3] = 'D'; // Godot Shader Binary Data.
```

Comment:
Are we going to need to do anything to ensure compatibility here? I guess this just invalidates all the old cache and shaders will have to be cached again, right?

Reply:
In D3D12 there's at least one case where the GLSL must be patched.

Reply:
I deleted my previous answer because I misread the question.

You're right in that there's no danger of clash since Vulkan and D3D12 will get different shader hashes.

Reply:
Also, the commit hash participates in the shader cache key hash, too, so there's also no concern with already cached Vulkan shaders.

Reply:
Sounds great!